pkg/streamingpromql/engine_test.go

```diff
@@ -273,6 +283,279 @@ func TestRangeVectorSelectors(t *testing.T) {
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "histogram_with_gaps"),
+						Histograms: []promql.HPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   2,
+									Count: 2,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 3,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 1, 0,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"histogram: metric with stale marker": {
+			expr: "histogram_with_stale_marker[3m]",
+			ts:   baseT.Add(3 * time.Minute),
+			expected: &promql.Result{
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "histogram_with_stale_marker"),
+						Histograms: []promql.HPoint{
+							{
+								T: timestamp.FromTime(baseT),
+								H: &histogram.FloatHistogram{
+									Sum:   1,
+									Count: 1,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 2,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 0,
+									},
+								},
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   2,
+									Count: 2,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 2,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 1,
+									},
+								},
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(3 * time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   4,
+									Count: 4,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 4,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 1, 1, 1,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"mixed series with histograms and floats": {
+			expr: "mixed_metric[4m]",
+			ts:   baseT.Add(4 * time.Minute),
+			expected: &promql.Result{
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "mixed_metric"),
+						Floats: []promql.FPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								F: 1,
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(2 * time.Minute)),
+								F: 2,
+							},
+						},
+						Histograms: []promql.HPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(3 * time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   3,
+									Count: 3,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 3,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 2, 1,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"mixed series with a float following a histogram causing lookback": {
```

Comment:
This description is a bit misleading - lookback doesn't apply to range vector selectors.

I'd also suggest rephrasing this so the words are in the same order as the points:

```suggestion
		"mixed series with a histogram then a float": {
```